sdk/tests/iot/adu/test_az_iot_adu.c

```diff
@@ -457,6 +453,20 @@ static uint8_t file_url_contoso[]
       "westus2/contoso-adu-instance--contoso-adu/9f9bdc01a5cd49c09e79e35505a913c5/"
       "contoso-v1.1.bin";
 
+static int setup(void** state)
+{
+  (void)state;
+
+  adu_device_properties = az_iot_adu_client_device_properties_default();
+  adu_device_properties.manufacturer = AZ_SPAN_FROM_STR(TEST_ADU_DEVICE_MANUFACTURER);
+  adu_device_properties.model = AZ_SPAN_FROM_STR(TEST_ADU_DEVICE_MODEL);
+  adu_device_properties.adu_version = AZ_SPAN_FROM_STR(TEST_AZ_IOT_ADU_CLIENT_AGENT_VERSION);
+  adu_device_properties.delivery_optimization_agent_version = AZ_SPAN_EMPTY;
```

Comment:
Should this be AZ_SPAN_LITERAL_EMPTY? Looks like that's what it used to be and what we set it to in the device_properties_default

Reply:
The `_LITERAL` variants of span initializers are used for setting values in struct literals. So anything with some 
```c
my_struct = { .myvalue = blah };
```
kind of initialization.

For assigning individual values of a struct, we can set using the non-literal version which adds a cast to that value.